drivers/rptun/rptun.c

```diff
@@ -608,6 +551,71 @@ static int rptun_ioctl(FAR struct rpmsg_s *rpmsg, int cmd, unsigned long arg)
   return ret;
 }
 
+static void rptun_panic_(FAR struct rpmsg_s *rpmsg)
+{
+  FAR struct rptun_priv_s *priv = (FAR struct rptun_priv_s *)rpmsg;
+
+  RPTUN_PANIC(priv->dev);
+}
+
+static void rptun_dump(FAR struct rpmsg_s *rpmsg)
+{
+  FAR struct rptun_priv_s *priv = (FAR struct rptun_priv_s *)rpmsg;
+  FAR struct rpmsg_virtio_device *rvdev = &priv->rvdev;
+  FAR struct rpmsg_device *rdev = rpmsg->rdev;
+  FAR struct rpmsg_endpoint *ept;
+  FAR struct metal_list *node;
+  bool needlock = true;
+
+  if (!rvdev->vdev)
+    {
+      return;
+    }
+
+  if (up_interrupt_context() || sched_idletask() ||
+      nxmutex_is_hold(&rdev->lock))
+    {
+      needlock = false;
+    }
+
+  if (needlock)
+    {
+      metal_mutex_acquire(&rdev->lock);
+    }
+
+  metal_log(METAL_LOG_EMERGENCY,
+            "Dump rpmsg info between cpu (master: %s)%s <==> %s:\n",
+            rpmsg_virtio_get_role(rvdev) == RPMSG_HOST ? "yes" : "no",
+            CONFIG_RPMSG_LOCAL_CPUNAME, rpmsg_get_cpuname(rdev));
+
+  metal_log(METAL_LOG_EMERGENCY, "rpmsg vq RX:\n");
```

Comment:
Replace all metal_log to rpmsg log
```suggestion
  rpmsgwarn("rpmsg vq RX:\n");
```

Reply:
let's create a new patch to do this?

Reply:
@anchao The `virtqueue_dump()` also use `metal_log()`, it's better to keep consistence, so do this change in next patch: re-implement the `virtqueue_dump()` and change all the `metal_log()` to `syslog()` or `rpmsgxxx()`?
Another question is should we change all `metal_xxx` to nuttx implemented version, such as `metal_list`?
Do you have some suggestions?